docs/api.md

```diff
@@ -181,3 +181,26 @@ Example (generated using the body and public key from the previous examples):
     }
 }
 ```
+
+## Authenticated Media
+
+When accessing media from a Synapse homeserver with authenticated media enabled, an
+`Authorization` HTTP header must be passed along with any request to the Matrix Content Scanner.
+If the `Authorization` header is not present, the content scanner assumes the request is not an
+authenticated media request and will use the old Matrix endpoints.
+
+This header follows the `Authentication Bearer scheme` as [outlined in the Matrix specification](https://spec.matrix.org/v1.12/client-server-api/#using-access-tokens).
+The `access_token` must be the Matrix access token of the client's user.
+The `Authorization` header method must be used, sending the access token as a query string
+parameter is not supported.
+
+Example authorization header:
+
+```
+Authorization: Bearer <access_token>
+```
+
+If a request is made for authenticated media and the access token is invalid, the content scanner
+will responde with a `MCS_MEDIA_REQUEST_FAILED`.
```

Comment:
```suggestion
will respond with HTTP status 502, errcode `MCS_MEDIA_REQUEST_FAILED`.
```

ref: https://github.com/element-hq/matrix-content-scanner-python/blob/694914f92bf100a37f7e1baee527cfb3cd01303f/src/matrix_content_scanner/scanner/file_downloader.py#L210-L231

Seems like we should probably be returning a 401 here though, in line with other Matrix-esque endpoints? (This doesn't need to be fixed here, but if you agree then we can track this in a separate issue.)

Reply:
Yup yup - this should be changed to be more helpful for clients to understand what is going wrong.
Issue: https://github.com/element-hq/matrix-content-scanner-python/issues/72